library/Fission/Storage/IPFS/Add.hs

```diff
@@ -0,0 +1,167 @@
+module Fission.Storage.IPFS.Add
+  ( addRaw
+  , addFile
+  , addPath
+  , addDir
+  ) where
+
+import           Data.ByteString.Lazy.Char8 as CL
+import           Data.List                  as List
+
+import qualified Network.HTTP.Client  as HTTP
+import qualified System.FilePath.Glob as Glob
+
+import           RIO.Directory
+import           RIO.FilePath
+import qualified RIO.ByteString.Lazy as Lazy
+
+import           Fission.Prelude
+import           Fission.Internal.Process
+import qualified Fission.Internal.UTF8 as UTF8
+
+import qualified Fission.IPFS.Process        as IPFS.Proc
+import           Fission.IPFS.Error          as IPFS.Error
+import           Fission.IPFS.Types          as IPFS
+import           Fission.IPFS.DAG.Node.Types as DAG
+import           Fission.IPFS.DAG.Link       as DAG.Link
+
+import           Fission.Storage.IPFS.DAG as DAG
+import qualified Fission.Storage.IPFS.Pin as IPFS.Pin
+
+addRaw ::
```

Comment:
Not sure if people saw my msg on discord about the syntax. I decided to write this PR with `::` on the same line as the function name. This fixes my syntax highlighting issues, but I also just think it's clearer because it's immediately apparent that that line is the start of a function type. I figured I'd do it with this PR to see how everyone felt about it (and make my life easier :wink:). Not sure how strong of convention this is, so let me know how you feel about it!

Reply:
The convention isn't super strong / just emerging. This is fine IMO 👍